### PR TITLE
Add stenv.cc.cc.lib to fix loading shared libraries: libstdc++.so.6

### PR DESCRIPTION
**Pull request by commandodev**
## Description of changes

Got the windmill executable building so it can execute without `error while loading shared libraries: libstdc++.so.6: cannot open shared object file: No such file or directory`

## Things done

Add `autoPatchelfIgnoreMissingDeps`  and `stdenv.cc.cc.lib` to buildInputs

- Built on platform(s)
  - [ x ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ x ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ x ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
I have the same fix in another PR, i didnt need the autopatchelfhook, can i ask what you needed it for ?

**Comment by commandodev** [comment]
Entirely possible I don't need that. Happy to remove my PR in favoir of
yours. Is there a way to signal boost and get one merged?

**Comment by happysalada** [comment]
If you remove the autopatchelf stuff , im happy to merge yours, mine wont be ready for a while.

Would you mind updating the package as well if possible?

**Comment by commandodev** [comment]
@happysalada I've made both of those changes. My plan ultimately is to add a `flake.nix` to windmill with a package and module.

**Comment by happysalada** [comment]
Since the repo is moving quite fast, that makes sense.
You can take whatever you find useful in the module i made. Im not done testing though as i still run into the postgres error

File: pkgs/servers/windmill/default.nix [hunk 1]
```diff
@@ -24,13 +24,15 @@
...
   pname = "windmill";
-  version = "1.188.1";
+  version = "1.200.0";
 
   fullSrc = fetchFromGitHub {
     owner = "windmill-labs";
     repo = "windmill";
     rev = "v${version}";
-    hash = "sha256-IiCIiP5KYRw10aPlR40RPW0ynXq5itf0LLtpDtxCNE4=";
+    # hash = "sha256-IiCIiP5KYRw10aPlR40RPW0ynXq5itf0LLtpDtxCNE4=";
```
Comment (happysalada) [inline pkgs/servers/windmill/default.nix:33]: do you need the commented lines ?

**Comment by happysalada** [comment]
@commandodev did you see my comments ?
If you are busy, I'm happy to take care of this in a separate PR, no worries, just let me know.

**Comment by 06kellyjac** [comment]
Additionally can the commit and PR title please be updated to match `CONTRIBUTING.md`, thanks

**Comment by commandodev** [comment]
Comments all noted, I'll make those changes this afternoon 👌

**Comment by happysalada** [comment]
Oh i missed that.
That would mean most likely one commit that would be like
```
windmill: 0.188.1 -> 0.200.0; fix build
Add necessary dynamic library 
```

Youll have to forgive me if the versions are not correct, im on my phone.

**Comment by happysalada** [comment]
Just so you know, I'm waiting for a fix that already hit master to be released but will probably merge this after https://github.com/NixOS/nixpkgs/pull/268259

**Comment by happysalada** [comment]
Should be fixed on master.

**Pull request closed without merging**